glib/src/gobject/type_module.rs

```diff
@@ -38,6 +38,21 @@ pub trait TypeModuleExt: IsA<TypeModule> + sealed::Sealed + 'static {
         }
     }
 
+    #[doc(alias = "g_type_module_register_enum")]
+    fn register_enum(
+        &self,
+        name: &str,
+        const_static_values: &'static [EnumValue],
```

Comment:
These functions all assume that the last `EnumValue` is the terminator, but that's not necessarily the case and if this is called otherwise you'll get a crash

Reply:
Or alternatively the function would have to be `unsafe` but that would seems unfortunate.

Maybe you could just add an assertion here for the last element being the terminator?

Reply:
Hm, actually we would panic when trying to get the name or nick of such a `EnumValue`, wouldn't we? Maybe we can have the macro create some wrapper type that ensures the terminator and when dereffing would give a `&[EnumValue]` without the terminator?

Reply:
Sounds good

Reply:
`EnumValues` should probably implement `Deref<Target = [EnumValue]>` so that a slice without the terminator is returned

Reply:
And maybe not re-exported from the crate root but make the `enum` module public instead. There should rarely be a reason for a user to type out the name, so having it outside the crate root reduces some noise

Reply:
Otherwise looks good to me, thanks :)